packages/web/src/ui/flow/design.js

```diff
@@ -479,16 +479,21 @@ const actions = ({ sources }) => {
           if (event.data instanceof Object) {
             const start = new Date()
 
-            /* FIXME: update later
-            const { CAG, CSG } = require('@jscad/csg')
             const solids = event.data.solids.map(function (object) {
               console.log('setting solids from worker', object)
-              if (object['class'] === 'CSG') { return CSG.fromCompactBinary(object) }
-              if (object['class'] === 'CAG') { return CAG.fromCompactBinary(object) }
-            })*/
-            const solids = event.data.solids.map(solid => JSON.parse(solid))
-            const { lookupCounts, lookup } = event.data 
-            
+              if (object[0] === 0) { // Geom2
```

Comment:
UG. This function should not know about the type of compact binary. Use the general function?

Reply:
yes it needs to know about it, since it is the one that is responsible for converting back from compact binary to distribute the data to the rest of the code :)